pennylane/qnode.py

```diff
@@ -559,45 +559,57 @@ def _best_method(self, idx):
         #
         # 5. Then run the standard discrete-case algorithm for determining the best gradient method
         # for every free parameter.
-        def best_for_op(o_idx):
+
+        # pylint: disable=too-many-return-statements
+        def best_for_op(op_node):
             "Returns the best gradient method for the operation op."
-            op = self.ops[o_idx]
             # for discrete operations, other ops do not affect the choice
+            o_idx = op_node["idx"]
+            op = op_node["op"]
+
             if not isinstance(op, pennylane.operation.CV):
                 return op.grad_method
 
             # for CV ops it is more complicated
-            if op.grad_method == 'A':
+            if op.grad_method == "A":
+                obs_successors = self._op_successors(o_idx, 'E')
+                if not obs_successors:
+                    # op is not succeeded by any observables, thus analytic method is OK
+                    return 'A'
+
                 # op is Gaussian and has the heisenberg_* methods
+                # A non-Gaussian successor is OK if it isn't succeeded by any observables
```

Comment:
These three comments don't really make sense read in order, do they need to be rewritten?

Reply:
Ok let me review these.